Update the readme (#331)
aviator-co/av: README.md

<div align="center">
  <img src="resources/isotype-color.svg" width="100">

>   <p><strong>av:</strong> Stacked PR manager</p>

Reviewer: ```suggestion
  <p><strong>av:</strong> CLI to manage Stacked PRs</p>
```
```suggestion
  <p><strong>av:</strong> Stacked PR manager</p>
```

</div>

---

> `av` is a command-line tool that helps you manage your PRs in a stacked way. It

Reviewer: ```suggestion
`av` is a command-line tool that helps you manage your stacked PRs on GitHub. It
```

> allows you to create a PR that depends on another PR, and it will automatically
> update the dependent PR when the base PR is updated.

Reviewer: ```suggestion
allows you to create a PR stacked on top of another PR, and it will
automatically update the dependent PR when the base PR is updated.
```

# Features

> - Create a PR that depends on another PR.

Reviewer: ```suggestion
- Create a PR that is stacked on another PR.
```

> - Visualize the PR stack.

Reviewer: ```suggestion
- Visualize the entire stack of PRs.
```

- Interactively navigate through the PR stack.
- Rebase the dependent PR when the base PR is updated.
- Remove the merged PRs from the stack.

> - Split a PR / commit into multiple PRs / commits.

Reviewer: ```suggestion
- Split a PR into multiple PRs.
- Split a commit into multiple commits.
```

previous PR:

```sh
$ av stack branch feature-2
$ echo "Another feature" >> hello.txt
$ git add hello.txt
$ git commit -m "Update hello.txt"
$ av pr create
```

Visualize the PR stack:

```sh
$ av stack tree
  * feature-2 (HEAD)
  │ https://github.com/octocat/Hello-World/pull/2
  │
  * feature-1
  │ https://github.com/octocat/Hello-World/pull/1
  │
  * master
```

Merge the first PR:

```sh
$ gh pr merge feature-1
```

Sync the stack:

```sh
$ av stack sync

  ✓ GitHub fetch is done
  ✓ Restack is done

    * ✓ feature-2 f9d85fe
    │
    * master 7fd1a60

  ✓ Pushed to GitHub

    Following branches do not need a push.

      feature-1: PR is already merged.

    Following branches are pushed.

      feature-2
        Remote: dbae4bd Update hello.txt 2024-06-11 16:41:18 -0700 -0700 (2 minutes ago)
        Local:  f9d85fe Update hello.txt 2024-06-11 16:43:41 -0700 -0700 (7 seconds ago)
        PR:     https://github.com/octocat/Hello-World/pull/2

  ✓ Deleted the merged branches

    Following merged branches are deleted.

      feature-1: f2335eec783b54226a7ab90f4af1c9b8309f8b61

```

# Installation

`av` is available for macOS and Linux. In order to interact with GitHub, `av`

> requires an API token. If you have [GitHub CLI](https://cli.github.com/)

Reviewer: ```suggestion
uses the GitHub API token. If you have [GitHub CLI](https://cli.github.com/)
```

> installed, `av` will use the token from the GitHub CLI. Recommended to install
> both.

Reviewer: ```suggestion
```suggestion
installed, `av` will use the token automatically from the GitHub CLI. It is
recommended to install both.
```

   ```sh
   av init
   ```

# Example commands

| Command               | Description                                                |
| --------------------- | ---------------------------------------------------------- |
| `av stack branch`     | Create a new child branch from the current branch.         |
| `av stack restack`    | Rebase the branches to their parents.                      |
| `av pr create`        | Create or update a PR.                                     |
| `av stack tree`       | Visualize the PRs.                                         |
| `av stack sync --all` | Fetch and rebase all branches.                             |
| `av stack adopt`      | Adopt a branch that is not created from `av stack branch`. |
| `av stack reparent`   | Change the parent of the current branch.                   |
| `av stack switch`     | Check out branches interactively.                          |
| `av stack reorder`    | Reorder the branches.                                      |
| `av commit amend`     | Amend the last commit and rebase the children.             |
| `av commit split`     | Split the last commit.                                     |

# How it works

> `av` internally keeps tracks of the PRs and their dependencies. For each branch,
> it remembers where the branch started. When the base branch is updated, `av`
> rebases the dependent branches on top of the new base branch using the
> remembered starting point as the merge base.

Reviewer: ```suggestion
`av` internally keeps tracks of the PRs, their associated branches and their dependent branches.
For each branch, it remembers where the branch started (the base commit of the branch). When
the base branch is updated, `av` rebases the dependent branches on top of the
new base branch using the remembered starting point as the merge base.
```
